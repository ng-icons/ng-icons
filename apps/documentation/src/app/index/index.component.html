<div
  appFadeInContainer
  class="flex flex-col items-center justify-center h-screen px-4 mt-6 md:mt-0"
>
  <img
    appFadeIn
    class="w-32 mb-6"
    src="assets/angular.svg"
    alt="Angular Logo"
  />

  <h2
    appFadeIn
    class="heading mb-5 text-base md:text-xl font-extrabold uppercase"
  >
    Angular Icons
  </h2>

  <h1
    appFadeIn
    class="slogan font-extrabold text-3xl md:text-5xl text-center mb-6 md:leading-[60px]"
  >
    The all-in-one icon<br />
    library for Angular
  </h1>

  <p appFadeIn class="text-sm md:text-base text-[#F5EAEA] text-center mb-16">
    Providing over 48,000 icons for use in your applications
  </p>

  <div appFadeIn class="flex gap-6 flex-col md:flex-row">
    <button
      routerLink="/getting-started"
      class="btn btn-white h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Get Started</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>

    <button
      routerLink="/browse-icons"
      class="btn btn-red h-12 px-6 flex gap-x-2 items-center font-medium text-sm transition-shadow ease-out duration-200"
    >
      <span class="mb-0.5">Browse Icons</span>

      <ng-icon name="heroChevronRightMini" class="text-xl"></ng-icon>
    </button>
  </div>
</div>
