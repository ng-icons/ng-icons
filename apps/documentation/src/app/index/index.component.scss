.heading {
  line-height: 25px;
  letter-spacing: 0.08em;
  background: linear-gradient(360deg, #ffafbd 0%, #ffc3a0 82%);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.slogan {
  line-height: 60px;
  background: linear-gradient(
    180deg,
    #ffffff 0%,
    rgba(227, 153, 153, 0.96) 100%
  );
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
  text-fill-color: transparent;
}

.btn {
  border-radius: 32px;
  box-shadow: 0 0 0 rgba(0, 0, 0, 0);

  &.btn-white {
    background: linear-gradient(90deg, #e0eafc 0%, #cfdef3 100%);
    border-top: 1px solid #ffffff;
    color: #1a1e2a;

    &:hover {
      box-shadow: 0px 1px 40px rgba(224, 234, 252, 0.4);
    }
  }

  &.btn-red {
    border-top: 1px solid rgba(238, 238, 238, 0.3);
    background: linear-gradient(90deg, #e53935 0%, #e35d5b 100%);
    color: #fff;

    &:hover {
      box-shadow: 0px 1px 40px rgba(216, 24, 47, 0.5);
    }
  }
}
