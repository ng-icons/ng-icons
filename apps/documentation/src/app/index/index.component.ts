import { Component } from '@angular/core';
import { RouterLink } from '@angular/router';
import { NgIconComponent, provideIcons } from '@ng-icons/core';
import { heroChevronRightMini } from '@ng-icons/heroicons/mini';
import { FadeInContainerDirective } from '../directives/fade-in/fade-in-container.directive';
import { FadeInDirective } from '../directives/fade-in/fade-in.directive';

@Component({
  selector: 'app-index',
  templateUrl: './index.component.html',
  styleUrls: ['./index.component.scss'],
  standalone: true,
  imports: [
    NgIconComponent,
    RouterLink,
    FadeInContainerDirective,
    FadeInDirective,
  ],
  providers: [provideIcons({ heroChevronRightMini })],
})
export class IndexComponent {}
