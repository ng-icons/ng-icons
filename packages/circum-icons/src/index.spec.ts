import * as icons from './index';

describe('CircumIcons Icons', () => {
  it('should ensure the export names have not changed unexpectedly', () => {
    expect(Object.keys(icons)).toMatchSnapshot();
  });
});
