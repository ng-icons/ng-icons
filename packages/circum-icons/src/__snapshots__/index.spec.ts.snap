// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircumIcons Icons should ensure the export names have not changed unexpectedly 1`] = `
Array [
  "circumAirportSign1",
  "circumAlarmOff",
  "circumAlarmOn",
  "circumAlignBottom",
  "circumAlignCenterH",
  "circumAlignCenterV",
  "circumAlignLeft",
  "circumAlignRight",
  "circumAlignTop",
  "circumApple",
  "circumAt",
  "circumAvocado",
  "circumBacon",
  "circumBadgeDollar",
  "circumBag1",
  "circumBandage",
  "circumBank",
  "circumBarcode",
  "circumBaseball",
  "circumBasketball",
  "circumBatteryCharging",
  "circumBatteryEmpty",
  "circumBatteryFull",
  "circumBeaker1",
  "circumBeerMugFull",
  "circumBellOff",
  "circumBellOn",
  "circumBezier",
  "circumBitcoin",
  "circumBluetooth",
  "circumBookmarkCheck",
  "circumBookmarkMinus",
  "circumBookmarkPlus",
  "circumBookmarkRemove",
  "circumBookmark",
  "circumBowlNoodles",
  "circumBoxList",
  "circumBoxes",
  "circumBrightnessDown",
  "circumBrightnessUp",
  "circumBullhorn",
  "circumBurger",
  "circumCalculator1",
  "circumCalculator2",
  "circumCalendarDate",
  "circumCalendar",
  "circumCamera",
  "circumChat1",
  "circumChat2",
  "circumCircleAlert",
  "circumCircleCheck",
  "circumCircleChevDown",
  "circumCircleChevLeft",
  "circumCircleChevRight",
  "circumCircleChevUp",
  "circumCircleInfo",
  "circumCircleList",
  "circumCircleMinus",
  "circumCircleMore",
  "circumCirclePlus",
  "circumCircleQuestion",
  "circumCircleRemove",
  "circumClock1",
  "circumClock2",
  "circumCloudDrizzle",
  "circumCloudMoon",
  "circumCloudOff",
  "circumCloudOn",
  "circumCloudRainbow",
  "circumCloudSun",
  "circumCloud",
  "circumCoffeeBean",
  "circumCoffeeCup",
  "circumCoinInsert",
  "circumCoins1",
  "circumCompass1",
  "circumCreditCard1",
  "circumCreditCard2",
  "circumCreditCardOff",
  "circumCrop",
  "circumDark",
  "circumDatabase",
  "circumDeliveryTruck",
  "circumDesktopMouse1",
  "circumDesktopMouse2",
  "circumDesktop",
  "circumDiscount1",
  "circumDollar",
  "circumDroplet",
  "circumDumbbell",
  "circumEdit",
  "circumEraser",
  "circumExport",
  "circumFaceFrown",
  "circumFaceMeh",
  "circumFaceSmile",
  "circumFacebook",
  "circumFileOff",
  "circumFileOn",
  "circumFilter",
  "circumFlag1",
  "circumFloppyDisk",
  "circumFolderOff",
  "circumFolderOn",
  "circumFootball",
  "circumForkKnife",
  "circumFries",
  "circumGift",
  "circumGlass",
  "circumGlobe",
  "circumGps",
  "circumGrid2H",
  "circumGrid2V",
  "circumGrid31",
  "circumGrid32",
  "circumGrid41",
  "circumGrid42",
  "circumHardDrive",
  "circumHashtag",
  "circumHeadphones",
  "circumHeart",
  "circumHome",
  "circumHospital1",
  "circumHotdog",
  "circumIceCream",
  "circumImageOff",
  "circumImageOn",
  "circumImport",
  "circumInboxIn",
  "circumInboxOut",
  "circumIndent",
  "circumInstagram",
  "circumKeyboard",
  "circumLaptop",
  "circumLemon",
  "circumLight",
  "circumLineHeight",
  "circumLink",
  "circumLinkedin",
  "circumLocationArrow1",
  "circumLocationOff",
  "circumLocationOn",
  "circumLock",
  "circumLogin",
  "circumLogout",
  "circumLollipop",
  "circumMail",
  "circumMapPin",
  "circumMap",
  "circumMaximize1",
  "circumMaximize2",
  "circumMedal",
  "circumMedicalCase",
  "circumMedicalClipboard",
  "circumMedicalCross",
  "circumMedicalMask",
  "circumMemoPad",
  "circumMicrochip",
  "circumMicrophoneOff",
  "circumMicrophoneOn",
  "circumMinimize1",
  "circumMinimize2",
  "circumMobile1",
  "circumMobile2",
  "circumMobile3",
  "circumMobile4",
  "circumMoneyBill",
  "circumMoneyCheck1",
  "circumMonitor",
  "circumMountain1",
  "circumMug1",
  "circumMusicNote1",
  "circumNoWaitingSign",
  "circumPalette",
  "circumPaperplane",
  "circumParking1",
  "circumPassport1",
  "circumPause1",
  "circumPen",
  "circumPenpot",
  "circumPercent",
  "circumPhone",
  "circumPickerEmpty",
  "circumPickerHalf",
  "circumPill",
  "circumPillsBottle1",
  "circumPizza",
  "circumPlane",
  "circumPlay1",
  "circumPlug1",
  "circumPower",
  "circumRainbow",
  "circumRead",
  "circumReceipt",
  "circumRedo",
  "circumRepeat",
  "circumRollingSuitcase",
  "circumRoute",
  "circumRouter",
  "circumRuler",
  "circumSatellite1",
  "circumSaveDown1",
  "circumSaveDown2",
  "circumSaveUp1",
  "circumSaveUp2",
  "circumSearch",
  "circumServer",
  "circumSettings",
  "circumShare1",
  "circumShare2",
  "circumShirt",
  "circumShop",
  "circumShoppingBasket",
  "circumShoppingCart",
  "circumShoppingTag",
  "circumShuffle",
  "circumSignpostDuo1",
  "circumSignpostL1",
  "circumSignpostR1",
  "circumSliderHorizontal",
  "circumSliderVertical",
  "circumSpeaker",
  "circumSquareAlert",
  "circumSquareCheck",
  "circumSquareChevDown",
  "circumSquareChevLeft",
  "circumSquareChevRight",
  "circumSquareChevUp",
  "circumSquareInfo",
  "circumSquareMinus",
  "circumSquareMore",
  "circumSquarePlus",
  "circumSquareQuestion",
  "circumSquareRemove",
  "circumStar",
  "circumStethoscope",
  "circumStickyNote",
  "circumStop1",
  "circumStopSign1",
  "circumStopwatch",
  "circumStreamOff",
  "circumStreamOn",
  "circumSun",
  "circumTablets1",
  "circumTempHigh",
  "circumTextAlignCenter",
  "circumTextAlignJustify",
  "circumTextAlignLeft",
  "circumTextAlignRight",
  "circumText",
  "circumTimer",
  "circumTrash",
  "circumTrophy",
  "circumTurnL1",
  "circumTurnR1",
  "circumTwitter",
  "circumUmbrella",
  "circumUndo",
  "circumUnlock",
  "circumUnread",
  "circumUsb",
  "circumUser",
  "circumVault",
  "circumVial",
  "circumVideoOff",
  "circumVideoOn",
  "circumViewBoard",
  "circumViewColumn",
  "circumViewList",
  "circumViewTable",
  "circumViewTimeline",
  "circumVirus",
  "circumVoicemail",
  "circumVolumeHigh",
  "circumVolumeMute",
  "circumVolume",
  "circumWallet",
  "circumWarning",
  "circumWavePulse1",
  "circumWheat",
  "circumWifiOff",
  "circumWifiOn",
  "circumYoutube",
  "circumZoomIn",
  "circumZoomOut",
]
`;
